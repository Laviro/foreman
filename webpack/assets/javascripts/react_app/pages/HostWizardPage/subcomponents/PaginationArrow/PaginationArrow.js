import React from 'react';
import PropTypes from 'prop-types';
import { Icon, noop } from 'patternfly-react';
import classNames from 'classnames';

const PaginationArrow = ({ direction, disabled, onClick }) => {
  const handleClick = () => {
    if (disabled) return;
    onClick();
  };
  return (
    <div
      className={classNames('pagination_arrow', { disabled })}
      onClick={handleClick}
    >
      <Icon name={`arrow-circle-${direction}`} />
    </div>
  );
};

PaginationArrow.propTypes = {
  direction: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
};

PaginationArrow.defaultProps = {
  direction: 'left',
  disabled: false,
  onClick: noop,
};

export default PaginationArrow;
