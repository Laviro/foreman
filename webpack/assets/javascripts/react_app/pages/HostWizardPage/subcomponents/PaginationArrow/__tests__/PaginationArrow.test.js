import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import PaginationArrow from '../PaginationArrow';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('PaginationArrow', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(PaginationArrow, fixtures));
});
