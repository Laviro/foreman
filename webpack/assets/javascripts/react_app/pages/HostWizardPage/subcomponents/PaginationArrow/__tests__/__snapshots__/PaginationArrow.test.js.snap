// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationArrow rendering render without Props 1`] = `
<div
  className="pagination_arrow"
  onClick={[Function]}
>
  <Icon
    name="arrow-circle-left"
    type="fa"
  />
</div>
`;
